src/firefly/js/templates/lightcurve/LcPeriodogram.jsx

```diff
@@ -505,7 +506,18 @@ function periodogramSuccess(popupId, hideDropDown = false) {
     return (request) => {
         const tbl = getTblById(LC.RAW_TABLE);
         const layoutInfo = getLayouInfo();
-        let srcFile = get(tbl, 'request.source');
+
+        let tblRequest = get(tbl, 'request');
+        if (tblRequest.filters){
+            tblRequest.pageSize = MAX_ROW;
+            const selectInfo=get(tbl, 'selectInfo');
+            if (selectInfo){
+                setSelectInfo(tblRequest, selectInfo)
+            }
+
+        }
+
+        let srcFile =  get(tbl, 'request.filters')? JSON.stringify(tblRequest):get(tbl, 'request.source');
```

Comment:
This should be done ALL the time and not just on filtered table.  Also, I don't understand the code above.
This is the only change I expect to see.
`let srcFile =  JSON.stringify(tbl.request);`

Reply:
I thought so.  But after I did this way, the saved table only has 100 rows.  I investigated the reason and found out that the selectInfo is needed to set the totalRowCount.  After I made the changes, I got the filtered table right.